### ISSUE TITLE
Throws "more than one nested type..." when multiple table-valued input parameters have the same user-defined table type

### ISSUE THREAD

**Issue by CalvinDale** [fsprojects/FSharp.Data.SqlClient#170]
I have defined a UDF whose parameters are all table-valued inputs. And several of these parameters are specified in terms of the same user-defined table type. When I try to construct an instance of one of the parameters whose types are shared with other parameters, I encounter this error:
```
The type provider 'FSharp.Data.SqlCommandProvider' reported an error: There is more than one nested type called '[SHARED_TYPE]' in type 'FSharp.Data.SqlCommandProvider,CommandText="[...]"
```
Here's the T-SQL code:
```
CREATE TYPE dbo.ConMetPartNumberTableType AS TABLE (PartNumber dbo.udt_ConMetPartNumber NOT NULL);
CREATE TYPE dbo.IntegerTableType          AS TABLE (Value INT          NOT NULL);
CREATE TYPE dbo.Decimal3x2TableType       AS TABLE (Value DECIMAL(3,2) NOT NULL);
CREATE TYPE dbo.Decimal4x2TableType       AS TABLE (Value DECIMAL(4,2) NOT NULL);
CREATE TYPE dbo.Decimal4x3TableType       AS TABLE (Value DECIMAL(4,3) NOT NULL);
CREATE TYPE dbo.Decimal5x0TableType       AS TABLE (Value DECIMAL(5,0) NOT NULL);
CREATE TYPE dbo.Decimal5x3TableType       AS TABLE (Value DECIMAL(5,3) NOT NULL);

CREATE FUNCTION BrakeDrum.ft_Filter_BrakeDrumNumbers (
	@BrakeDrumNumbers       dbo.ConMetPartNumberTableType READONLY,
	@SizeDiameters          dbo.Decimal4x2TableType       READONLY,
	@SizeWidths             dbo.Decimal5x3TableType       READONLY,
	@DrumTypes              dbo.IntegerTableType          READONLY,
	@PilotDiameters         dbo.Decimal4x2TableType       READONLY,
	@OverallDepths          dbo.Decimal4x2TableType       READONLY,
	@SurfaceWidths          dbo.Decimal4x2TableType       READONLY,
	@FlangeThicknesses      dbo.Decimal3x2TableType       READONLY,
	@BoltCircleDiameters    dbo.Decimal4x2TableType       READONLY,
	@BoltHoleDiameters      dbo.Decimal4x3TableType       READONLY,
	@WheelStudHoleCounts    dbo.IntegerTableType          READONLY
)
RETURNS TABLE
RETURN
	SELECT TOP 10 bd.BrakeDrumNumber, p.[Description]
	FROM BrakeDrum.BrakeDrums bd
	JOIN BrakeDrum.Sizes s ON bd.BrakeDrumSizeId = s.Id
	JOIN Part.Parts p on bd.BrakeDrumNumber = p.PartNumber
	WHERE ISNUMERIC(bd.BrakeDrumNumber) = 1
	  AND p.StatusId = 2 -- Public
	  AND (((SELECT COUNT(PartNumber) FROM @BrakeDrumNumbers         ) = 0) OR bd.BrakeDrumNumber        IN (SELECT PartNumber FROM @BrakeDrumNumbers   ))
	  AND (((SELECT COUNT(Value)      FROM @SizeDiameters            ) = 0) OR s.DiameterInch            IN (SELECT Value      FROM @SizeDiameters      ))
	  AND (((SELECT COUNT(Value)      FROM @SizeWidths               ) = 0) OR s.WidthInch               IN (SELECT Value      FROM @SizeWidths         ))
	  AND (((SELECT COUNT(Value)      FROM @DrumTypes                ) = 0) OR bd.BrakeDrumTypeId        IN (SELECT Value      FROM @DrumTypes          ))
	  AND (((SELECT COUNT(Value)      FROM @PilotDiameters           ) = 0) OR bd.PilotDiameterInch      IN (SELECT Value      FROM @PilotDiameters     ))
	  AND (((SELECT COUNT(Value)      FROM @OverallDepths            ) = 0) OR bd.DepthInch              IN (SELECT Value      FROM @OverallDepths      ))
	  AND (((SELECT COUNT(Value)      FROM @SurfaceWidths            ) = 0) OR bd.SurfaceWidthInch       IN (SELECT Value      FROM @SurfaceWidths      ))
	  AND (((SELECT COUNT(Value)      FROM @FlangeThicknesses        ) = 0) OR bd.ThicknessInch          IN (SELECT Value      FROM @FlangeThicknesses  ))
	  AND (((SELECT COUNT(Value)      FROM @BoltCircleDiameters      ) = 0) OR bd.BoltCircleDiameterInch IN (SELECT Value      FROM @BoltCircleDiameters))
	  AND (((SELECT COUNT(Value)      FROM @BoltHoleDiameters        ) = 0) OR bd.BoltHoleDiameterInch   IN (SELECT Value      FROM @BoltHoleDiameters  ))
	  AND (((SELECT COUNT(Value)      FROM @WheelStudHoleCounts      ) = 0) OR bd.WheelStudHoleCount     IN (SELECT Value      FROM @WheelStudHoleCounts))
	ORDER BY LEN(bd.BrakeDrumNumber), bd.BrakeDrumNumber
```
And here's the F# code:
```
let GetFilterBrakeDrumNumbers(brakeDrumNumbers : string[], sizeDiameters: decimal[]) = 
    let brakeDrumNumbersParam = brakeDrumNumbers |> Array.map (fun brakeDrumNumber ->  GetFilterBrakeDrumNumbers_SqlCommandProvider.ConMetPartNumberTableType(brakeDrumNumber))
    let sizeDiametersParam = sizeDiameters |> Array.map (fun sizeDiameter ->  GetFilterBrakeDrumNumbers_SqlCommandProvider.Decimal4x2TableType(sizeDiameter))
```
I am able to construct the first parameter `brakeDrumNumbersParam` since its type is not shared with other parameters. But second parameter `sizeDiametersParam` produces the error as its type `Decimal4x2TableType` is shared with four other parameters.

Is there a recommended approach to deal with this issue?

I'm sure that it is possible in this case to define all of the decimal parameters in terms of a larger-sized decimal or alternatively a float. But that approach would entail casting and rounding and possibly tolerance-comparing all values and would result in a slower query. Furthermore since these precision and scaled decimal types are used extensively throughout this database, I prefer not to have to cast and round inputs every time they're used. So it would be nice to figure out a better way to deal with them when using the `FSharp.Data.SqlClient`.

Thanks!

**Comment by dmitry-a-morozov** [comment]
Fixed in ver 1.7.7
Thanks for reporting the issue !

**Comment by CalvinDale** [comment]
Thanks for the quick turn-around!
